src/main/scala/com/mozilla/telemetry/streaming/ErrorAggregatorBase.scala

```diff
@@ -0,0 +1,296 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+package com.mozilla.telemetry.streaming
+
+import com.mozilla.telemetry.heka.{Dataset, Message}
+import com.mozilla.telemetry.pings.{CorePing, CrashPing, MainPing, Ping}
+import com.mozilla.telemetry.timeseries.{RowBuilder, SchemaBuilder}
+import org.apache.spark.sql.{DataFrame, Row, SparkSession}
+import org.apache.spark.sql.catalyst.encoders.RowEncoder
+import org.apache.spark.sql.functions.{col, sum, window}
+import org.apache.spark.sql.types.{BinaryType, StructField, StructType}
+import org.json4s.DefaultFormats
+import org.rogach.scallop.{ScallopConf, ScallopOption}
+
+abstract class ErrorAggregatorBase extends StreamingJobBase {
+
+  val kafkaTopic = "telemetry"
+  val defaultNumFiles = 60
+
+  //TODO: make relationships visible here - we're adding crash/Fennec and core/Fennec/Android
+  private val allowedDocTypes = List("main", "crash", "core")
+  private val allowedAppNames = List("Firefox", "Fennec")
+  private val coreFennecPingAllowedOses = List("Android")
+  private val disallowedChannels = List("Other")
+  private val kafkaCacheMaxCapacity = 1000
+
+  private class Opts(args: Array[String]) extends BaseOpts(args) {
+    val outputPath: ScallopOption[String] = opt[String](
+      "outputPath",
+      descr = "Output path",
+      required = false,
+      default = Some("/tmp/parquet"))
+    val raiseOnError: ScallopOption[Boolean] = opt[Boolean](
+      "raiseOnError",
+      descr = "Whether the program should exit on a data processing error or not.")
+    val failOnDataLoss: ScallopOption[Boolean] = opt[Boolean](
+      "failOnDataLoss",
+      descr = "Whether to fail the query when it’s possible that data is lost.")
+    val numParquetFiles: ScallopOption[Int] = opt[Int](
```

Comment:
This is the same as `fileLimit` in the `BaseOpts` class. Why don't we remove this argument in favor of `fileLimit` and move the `conflicts` to `BaseOpts` as well?

Reply:
These are not the same - `fileLimit` is used for reading data via `Dataset` api (I haven't used this, but I can imagine it being useful for testing), whereas `numParquetFiles` controls number of files written per partition (I've clarified it's description).